### PR TITLE
#24 set max_body_size

### PR DESCRIPTION
**Pull request by julianseeger**

### CODE REVIEW

File: src/blockserver/server.py [hunk 1]
```diff
@@ -103,6 +104,7 @@ def initialize(self, transfer_cls, get_auth_cls, get_cache_cls, database_pool,
         self.auth = None
         self.streamer = None
         await self._authorize_request()
+        self.request.connection.max_body_size = options.max_body_size
```
Comment (audax) [inline src/blockserver/server.py]: This is only needed for POST requests, so you can move it down one line.

**Comment by audax** [comment]
You can test this for example by setting the option to 1 byte and then testing if the new max body size is set. You can also check if a request with a very large Content-Length header is accepted and test if that is only possible for POST requests and not other types of requests or requests to other handlers.

**Comment by julianseeger** [comment]
Get and delete without body are not allowed and put is not implemented (at least it throws crazy errors).
Now it's tested that the option controlls the allowed body size and given correct configuration, is should work fine now.

**Comment by audax** [comment]
Do you mean "get and delete _with_ a body"?

**Comment by julianseeger** [comment]
indeed ;)

**Pull request merged**